patches/server/0962-Add-player-death-screen-message-API.patch

```diff
@@ -0,0 +1,53 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: dawon <dawon@dawon.eu>
+Date: Fri, 24 Feb 2023 15:21:21 +0100
+Subject: [PATCH] Add player death screen message API
+
+
+diff --git a/src/main/java/net/minecraft/server/level/ServerPlayer.java b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+index 9160725f031c9698a00229c3dfa19d39b4a826c1..f91add6c8d21fd84a4b9f4c40c4f8c62cc5361d0 100644
+--- a/src/main/java/net/minecraft/server/level/ServerPlayer.java
++++ b/src/main/java/net/minecraft/server/level/ServerPlayer.java
+@@ -923,12 +923,15 @@ public class ServerPlayer extends Player {
+ 
+         net.kyori.adventure.text.Component deathMessage = event.deathMessage() != null ? event.deathMessage() : net.kyori.adventure.text.Component.empty(); // Paper - Adventure
+ 
+-        if (deathMessage != null && deathMessage != net.kyori.adventure.text.Component.empty() && flag) { // Paper - Adventure // TODO: allow plugins to override?
+-            Component ichatbasecomponent = PaperAdventure.asVanilla(deathMessage); // Paper - Adventure
+-
+-            this.connection.send(new ClientboundPlayerCombatKillPacket(this.getCombatTracker(), ichatbasecomponent), PacketSendListener.exceptionallySend(() -> {
++        // Paper start - death screen message api
++        net.kyori.adventure.text.Component deathScreenMessage = event.deathScreenMessage() != null ? event.deathScreenMessage() : deathMessage;
++        if ((true || deathMessage != null && deathMessage != net.kyori.adventure.text.Component.empty()) && flag) { // Paper - Adventure
```

Comment:
Why is this true here?

Reply:
`boolean hasDeathMessage = deathMessage != null && deathMessage != net.kyori.adventure.text.Component.empty();
boolean hasDeathScreenMessage = deathScreenMessage != null && deathScreenMessage != net.kyori.adventure.text.Component.empty();

if ((hasDeathMessage || hasDeathScreenMessage) && flag) {
    Component ichatbasecomponent = hasDeathScreenMessage ? PaperAdventure.asVanilla(deathScreenMessage) : PaperAdventure.asVanilla(deathMessage);`
    
Maybe something like this?

Reply:
> Why is this true here?
To minimise the Diff normally

Reply:
> ```
> boolean hasDeathMessage = deathMessage != null && deathMessage != net.kyori.adventure.text.Component.empty();
> boolean hasDeathScreenMessage = deathScreenMessage != null && deathScreenMessage != net.kyori.adventure.text.Component.empty();
> 
> if ((hasDeathMessage || hasDeathScreenMessage) && flag) {
>     Component ichatbasecomponent = hasDeathScreenMessage ? PaperAdventure.asVanilla(deathScreenMessage) : PaperAdventure.asVanilla(deathMessage);
> ```
> 
> Maybe something like this?

nvm im stupid lol

Reply:
You it is there to minimize the diff. I have moved this if but below... :-)